core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -17,11 +21,13 @@
 /**
  * Encloses classes related to the compression and decompression of messages.
  */
+@ExperimentalApi
 public final class MessageEncoding {
   /**
    * Special sentinel codec indicating that no compression should be used.  Users should use
    * reference equality to see if compression is disabled.
    */
+  @ExperimentalApi
```

Comment:
Where do we do `==` on the codec? Maybe it's not really a use case we care that much about.